src/main/update/check.ts

```diff
@@ -0,0 +1,112 @@
+import { IApv, ISimpleApv } from "src/interfaces/apv";
+import { IDownloadUrls, getDownloadUrls } from "../../utils/url";
+import Headless from "../headless/headless";
+
+export class GetPeersApvFailedError extends Error {}
+
+export interface IUpdateContext {
+  newApv: ISimpleApv;
+  oldApv: ISimpleApv;
+  urls: IDownloadUrls;
+}
+
+export async function checkUpdateRequired(
+  standalone: Headless,
+  platform: NodeJS.Platform,
+  // TODO: should make config object
+  netenv: string,
```

Comment:
Could we get this directly from the config instead?

Reply:
[Done](https://github.com/planetarium/9c-launcher/pull/1868#discussion_r969343016)